agent/04_agent_prepare_release.sh

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+
+LOGDIR=${SCRIPTDIR}/logs
+source $SCRIPTDIR/logging.sh
+source $SCRIPTDIR/common.sh
+source $SCRIPTDIR/network.sh
+source $SCRIPTDIR/utils.sh
+source $SCRIPTDIR/validation.sh
+source $SCRIPTDIR/agent/common.sh
+source $SCRIPTDIR/ocp_install_env.sh
+
+# To replace an image entry in the openshift release image, set <ENTRYNAME>_LOCAL_REPO so that:
+# - ENTRYNAME matches an uppercase version of the name in the release image with "-" converted to "_" 
+# - The var value must point to an already locally cloned repo
+#
+# To specify a custom Dockerfile set <ENTRYNAME>_DOCKERFILE, as a relative path of the Dockerfile
+# within the configured repo
+#
+# For example, to use a custom installer image:
+# export INSTALLER_LOCAL_REPO=~/go/src/github.com/openshift/installer
+# export INSTALLER_DOCKERFILE=images/installer/Dockerfile.ci
+
+early_deploy_validation
+write_pull_secret
+
+# Release mirroring could be required by the subsequent steps
+# even if the current one will be skipped
+if [[ ! -z "${MIRROR_IMAGES}" ]]; then
+   setup_release_mirror
```

Comment:
I would expect. particularly for REGISTRY_BACKEND=quay that would need to call `setup_local_registry` first. I think for podman backend that it will be called in 02_configure_host.sh

Reply:
The local registry is a sort of global shared resource that may be needed by different steps in the workflow. I think we should try to keep having, also for the sake of simplicity, an (early) single point of instantiation for the local registry regardless of the underlying backend (and the `02_configure_host.sh` seems a good fit).
Also mirroring the release it's something that could be required by different steps, so that's why I moved it here up in the chain